differs/pip_diff.go

```diff
@@ -92,18 +94,42 @@ func (a PipAnalyzer) getPackages(image pkgutil.Image) (map[string]map[string]uti
 				packageName := packageMatch[1]
 				version := packageMatch[2][:len(packageMatch[2])-1]
 
-				// Retrieves size for actual package/script corresponding to each dist-info metadata directory
-				// by taking the file entry alphabetically before it (for a package) or after it (for a script)
+				// First, try and use the "top_level.txt",
+				// Many egg packages contains a "top_level.txt" file describing the directories containing the
```

Comment:
https://setuptools.readthedocs.io/en/latest/formats.html

> The minimum project metadata that all eggs must have is a standard Python PKG-INFO file, named PKG-INFO and placed within the metadata directory appropriate to the format. 
...
> In addition to the PKG-INFO file, an egg’s metadata directory may also include files and directories representing various forms of optional standard metadata  ...

From the sound of it, `PKG-INFO` might be a more reliable choice than the `top_level.txt` which is optional

Reply:
Yeah, eggs are pretty oldschool/inconsistent/a bit horrid.
No worries, maybe not as bulletproof, but reading `top_level.txt` is still an improvement and as you say, if it gives you extra information needed - it sounds like a good pragmatic choice.